src/Razor/src/Microsoft.AspNetCore.Razor.OmniSharpPlugin/DefaultProjectChangePublisher.cs

```diff
@@ -94,7 +95,8 @@ protected virtual void SerializeToFile(OmniSharpProjectSnapshot projectSnapshot,
         // by the time we move the tempfile into its place
         using (var writer = tempFileInfo.CreateText())
         {
-            _serializer.Serialize(writer, projectSnapshot);
+            var projectRazorJson = new ProjectRazorJson(publishFilePath, projectSnapshot.InternalProjectSnapshot);
+            _serializer.Serialize(writer, projectRazorJson);
```

Comment:
The easiest fix here is for `OmniSharpProjectSnapshot` to have a `Serialize` method, that takes in `_serializer_ and `publishFilePath`.